utopia-remix/app/handlers/validators.ts

```diff
@@ -0,0 +1,37 @@
+import { AccessValidator, ensure, getUser } from '../util/api.server'
+import { UserProjectPermission } from '../types'
+import { Params } from '@remix-run/react'
+import { Status } from '../util/statusCodes'
+import { hasUserProjectPermission } from '../services/permissionsService.server'
+import { getProjectById } from '../models/project.server'
+
+export function validateProjectAccess(
+  permission: UserProjectPermission,
+  {
+    errorMessage,
+    status,
+  }: {
+    errorMessage?: string
+    status?: number
+  } = {},
+): AccessValidator {
+  return async function (req: Request, params: Params<string>) {
+    const { id: projectId } = params
+    ensure(projectId != null, 'project id is null', Status.BAD_REQUEST)
+
+    const projectData = await getProjectById({ id: projectId })
```

Comment:
we don't need the whole project data, only the ownership data, so the related function should only get that field from the db (and be renamed accordingly)

Reply:
here too please (https://github.com/concrete-utopia/utopia/pull/4929#discussion_r1514195378)